test/jdk/java/lang/invoke/BigArityTest.java

```diff
@@ -24,7 +24,18 @@
 /* @test
  * @summary High arity invocations
  * @compile BigArityTest.java
- * @run junit/othervm/timeout=2500 -XX:+IgnoreUnrecognizedVMOptions -XX:-VerifyDependencies -XX:CompileCommand=memlimit,*.*,0 -esa -DBigArityTest.ITERATION_COUNT=1 test.java.lang.invoke.BigArityTest
+ * @comment The reduced maximum node limit below avoids, in combination with
+ *          -Xcomp, timeouts and excessive C2 memory consumption during
+ *          compilation of some methods that both
+ *            (1) have a large number of parameters, and
+ *            (2) use JSR292 methods internally (which increases the
+ *                MaxNodeLimit with a factor of 3)
+ * @run junit/othervm/timeout=2500 -XX:+IgnoreUnrecognizedVMOptions
+ *                                 -XX:MaxNodeLimit=20000
+ *                                 -XX:-VerifyDependencies
+ *                                 -XX:CompileCommand=memlimit,*.*,0
+ *                                 -esa -DBigArityTest.ITERATION_COUNT=1
+ *                                 test.java.lang.invoke.BigArityTest
```

Comment:
Would it make sense to also have a run with fewer flags?

Reply:
Or why do we need to set all these flags? If we need some of them, then a comment could be helpful.

Reply:
Note that I only added the `-XX:MaxNodeLimit=20000`, all the other flags are from before (I just added line breaks). It could very well make sense to have a run with fewer flags, but I'm not sure if that's compatible with what the original author intended. I'd prefer leaving it as it is.

Reply:
Is it the use of the JSR292 methods that increases the limit? Or do you need to increase the limit to make sure we don't hit the limit?

Reply:
Yes, the use of JSR292 methods increases the `MaxNodeLimit`:
```
  // Bump max node limit for JSR292 users
  if (bc() == Bytecodes::_invokedynamic || orig_callee->is_method_handle_intrinsic()) {
    C->set_max_node_limit(3*MaxNodeLimit);
  }
```